cmd/template/dbdriver/mongo.go

```diff
@@ -0,0 +1,52 @@
+package dbdriver
+
+type MongoTemplate struct{}
+
+func (m MongoTemplate) Service() []byte {
+	return []byte(`package database
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Service interface {
+	Health() map[string]string
+}
+
+type service struct {
+	db *mongo.Client
+}
+
```

Comment:
nit:
maybe we include this in the existing Server struct?

Reply:
I've just tried this approach, and it is possible to implement, although it would require us to change the server template by making probably more versions of it depending on the chosen DB driver, while currently, we are only depending on the database package and making some additions to the routes template when the user chooses a driver to get installed, while the server package remains untouched.

I would like to know your take on this. If you think I might have missed your point,

Reply:
you mean something like this 

```go
package server

type Server struct {
	port int
	database.Service
}


And then use it directly in the Server template to create a new instance and call the HealthHandler method?

Reply:
Hmm, that's right. I will try to tweak the implementation tonight.

Reply:
something like this maybe ? 

```go

type Server struct {
	port int
	db   *database.Service
}


func NewServer() *http.Server {

	NewServer := &Server{
		port: port,
		db:   database.New(),
	}

	// Declare Server config
	server := &http.Server{
		Addr:         fmt.Sprintf(":%d", NewServer.port),
		Handler:      NewServer.RegisterRoutes(),
		IdleTimeout:  time.Minute,
		ReadTimeout:  10 * time.Second,
		WriteTimeout: 30 * time.Second,
	}

	return server
}
```

then using it in the route package like this 

```go
func (s *Server) RegisterRoutes() http.Handler {
	r := gin.Default()
	r.GET("/", s.helloWorldHandler)
	r.GET("/health", s.healthHandler)

	return r
}


func (s *Server) healthHandler(c *gin.Context) {
	c.JSON(http.StatusOK, s.db.Health())
}
```

Reply:
Yeah - this is exactly what I was thinking of as well.
However, like in my previous comment, if we do not have an actual DB spun up for the user then
```r.GET("/health", s.healthHandler)```
Will error out if they try to cURL if after running `make run`

Reply:
Will work on the new implementation for all the drivers asap? Also, I've answered your question in the earlier convo